.github/workflows/callable-e2e-runner.yml

```diff
@@ -0,0 +1,47 @@
+name: 'E2E Test Runner'
+
+on:
+  workflow_call:
+
+jobs:
+  e2e-prep:
+    name: Get required configs to run e2e tests
+    runs-on: ubuntu-latest
+    outputs:
+      integ-config: ${{env.INTEG_CONFIG}}
+      integ-utils: ${{env.INTEG_UTILS}}
+    steps:
+      - name: checkout AmplifyJs
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f
+        with:
+          path: amplify-js
+      - name: read integ config files
+        run: |
+          echo "INTEG_CONFIG=$(cat .github/integ-config/integ-*.yml | yq -o=json | jq -c .)" >> $GITHUB_ENV
+          echo "INTEG_UTILS=$(cat .github/integ-config/utils.yml | yq -o=json | jq -c .)" >> $GITHUB_ENV
+        working-directory: ./amplify-js
+
+  e2e-test-runner:
+    name: e2e test runnner
+    needs: e2e-prep
+    secrets: inherit  
+    strategy:
+      matrix:
+        integ-config: ${{ fromJson(needs.e2e-prep.outputs.integ-config) }}
+    uses: ./.github/workflows/callable-e2e-tests.yml
+    with:
+      test_name: ${{ matrix.integ-config.test_name }}
+      framework: ${{ matrix.integ-config.framework }}
+      category: ${{ matrix.integ-config.category }}
+      sample_name: ${{ matrix.integ-config.sample_name }}
+      spec: ${{ matrix.integ-config.spec }}
+      browser: |
+        ${{
+          toJSON(
+            fromJson(
+              needs.e2e-prep.outputs.integ-utils
+            )[matrix.integ-config.browser]
+          )
```

Comment:
Hard for me to parse what env this is running in. Its a place where a bit of code golf could be fun. Like, can we use jq or some other util to cheat this down to fewer characters.

Since this is working, would vote we leave it. Can iterate later.  We'll need to give each of these a think through security eyes to make sure we're not leaking secrets or access exposure through any of these interpolation. Will need the same thinking for my category list script.

No action here. Just sharing thoughts it makes me have.

Reply:
yes, makes sense
I'll make not of it